runtime/lua/vim/lsp/protocol.lua

```diff
@@ -318,6 +318,96 @@ for k, v in pairs(constants) do
   protocol[k] = tbl
 end
 
+--Lsp methods (based on version 3.18)
+protocol.Methods = {
+  Initialized = 'initialized',
+  Exit = 'exit',
+  ShutDown = 'shutdown',
+  SetTrace = '$/setTrace',
+  LogTrace = '$/logTrace',
+  ClientRegisterCapability = 'client/registerCapability',
+  ClientUnregisterCapability = 'client/unregisterCapability',
+  LogMessage = 'window/logMessage',
+  CancelRequest = '$/cancelRequest',
+  ShowMessage = 'window/showMessage',
+  ShowMessageRequest = 'window/showMesageRequest',
+  TelemetryEvent = 'telemetry/event',
+  DidOpen = 'textDocument/didOpen',
+  DidClose = 'textDocument/didClose',
+  WillSave = 'textDocument/willSave',
+  WillSaveWaitUnitl = 'textDocument/willSaveWaitUntil',
+  DidSave = 'textDocument/didSave',
+  DidChange = 'textDocument/didChange',
+  Hover = 'textDocument/hover',
+  Rename = 'textDocument/rename',
+  PrepareRename = 'textDocument/prepareRename',
+  References = 'textDocument/references',
+  Definition = 'textDocument/definition',
+  TypeDefinition = 'textDocument/typeDefinition',
+  Declaration = 'textDocument/declaration',
```

Comment:
why aren't the namespaces included in the symbol names? Again this harms discoverability by adding useless differences.

```suggestion
  textDocument_declaration = 'textDocument/declaration',
```